agent-inject/agent/container_sidecar_test.go

```diff
@@ -1275,3 +1276,220 @@ func TestContainerCache(t *testing.T) {
 		})
 	}
 }
+
+func TestAgentJsonPatch(t *testing.T) {
+
+	baseContainer := corev1.Container{
+		Name:    "vault-agent",
+		Image:   "foobar-image",
+		Command: []string{"/bin/sh", "-ec"},
+		Args:    []string{`echo ${VAULT_CONFIG?} | base64 -d > /home/vault/config.json && vault agent -config=/home/vault/config.json`},
+		Env: []v1.EnvVar{
+			{Name: "VAULT_LOG_LEVEL", Value: "info"},
+			{Name: "VAULT_LOG_FORMAT", Value: "standard"},
+			{Name: "VAULT_CONFIG", Value: "eyJhdXRvX2F1dGgiOnsibWV0aG9kIjp7InR5cGUiOiJrdWJlcm5ldGVzIiwibW91bnRfcGF0aCI6InRlc3QiLCJjb25maWciOnsicm9sZSI6InJvbGUiLCJ0b2tlbl9wYXRoIjoic2VydmljZWFjY291bnQvc29tZXdoZXJlL3Rva2VuIn19LCJzaW5rIjpbeyJ0eXBlIjoiZmlsZSIsImNvbmZpZyI6eyJwYXRoIjoiL2hvbWUvdmF1bHQvLnZhdWx0LXRva2VuIn19XX0sImV4aXRfYWZ0ZXJfYXV0aCI6ZmFsc2UsInBpZF9maWxlIjoiL2hvbWUvdmF1bHQvLnBpZCIsInZhdWx0Ijp7ImFkZHJlc3MiOiJodHRwOi8vZm9vYmFyOjEyMzQifSwidGVtcGxhdGVfY29uZmlnIjp7ImV4aXRfb25fcmV0cnlfZmFpbHVyZSI6dHJ1ZX19"},
+		},
+		Resources: v1.ResourceRequirements{
+			Limits:   v1.ResourceList{"cpu": resource.MustParse("500m"), "memory": resource.MustParse("128Mi")},
+			Requests: v1.ResourceList{"cpu": resource.MustParse("250m"), "memory": resource.MustParse("64Mi")},
+		},
+		VolumeMounts: []v1.VolumeMount{
+			{Name: "foobar", ReadOnly: true, MountPath: "serviceaccount/somewhere"},
+			{Name: "home-sidecar", MountPath: "/home/vault"},
+			{Name: "vault-secrets", MountPath: "/vault/secrets"},
+		},
+		Lifecycle: &v1.Lifecycle{
+			PreStop: &v1.Handler{
+				Exec: &v1.ExecAction{
+					Command: []string{"/bin/sh", "-c", "/bin/sleep 5 && /bin/vault token revoke -address=http://foobar:1234 -self"},
+				},
+			},
+		},
+		SecurityContext: &v1.SecurityContext{
+			Capabilities: &v1.Capabilities{
+				Drop: []v1.Capability{"ALL"},
+			},
+			RunAsGroup:               optional[int64](100),
+			RunAsUser:                optional[int64](1000),
+			RunAsNonRoot:             optional[bool](true),
+			ReadOnlyRootFilesystem:   optional[bool](true),
+			AllowPrivilegeEscalation: optional[bool](false),
+		},
+	}
+
+	baseInitContainer := corev1.Container{
```

Comment:
Since this is almost identical to baseContainer, we could maybe make a copy and then change the fields that are different, instead of redefining it all.

Reply:
:+1: